Mathlib/CategoryTheory/LiftingProperties/Limits.lean

```diff
@@ -0,0 +1,105 @@
+/-
+Copyright (c) 2024 Jack McKoen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jack McKoen
+-/
+import Mathlib.CategoryTheory.Limits.Shapes.Pullback.CommSq
+
+/-!
+# Lifting properties and (co)limits
+
+In this file, we show some consequences of lifting properties in the presence of
+certain (co)limits.
+
+-/
+
+universe v
+
+namespace CategoryTheory
+
+open Category Limits
+
+variable {C : Type*} [Category C] {X Y Z W : C}
+  {f : X ⟶ Y} {s : X ⟶ Z} {g : Z ⟶ W} {t : Y ⟶ W}
+
+lemma IsPushout.HasLiftingProperty (h : IsPushout s f g t)
+    {Z' W' : C} (g' : Z' ⟶ W') [HasLiftingProperty f g'] : HasLiftingProperty g g' where
+  sq_hasLift := fun {u v} sq ↦ by
+    have w : (s ≫ u) ≫ g' = f ≫ (t ≫ v) := by
+      rw [← Category.assoc, ← h.w, Category.assoc, Category.assoc, sq.w]
+    exact ⟨h.desc u (CommSq.mk w).lift (by rw [CommSq.fac_left]), h.inl_desc ..,
+      h.hom_ext (by rw [h.inl_desc_assoc, sq.w]) (by rw [h.inr_desc_assoc, CommSq.fac_right])⟩
+
+lemma IsPullback.HasLiftingProperty (h : IsPullback s f g t)
+    {X' Y' : C} (f' : X' ⟶ Y') [HasLiftingProperty f' g] : HasLiftingProperty f' f where
+  sq_hasLift := fun {u v} sq ↦ by
+    have w : (u ≫ s) ≫ g = f' ≫ v ≫ t := by
+      rw [Category.assoc, h.toCommSq.w, ← Category.assoc, ← Category.assoc, sq.w]
+    exact ⟨h.lift (CommSq.mk w).lift v (by rw [CommSq.fac_right]),
+      h.hom_ext (by rw [Category.assoc, h.lift_fst, CommSq.fac_left])
+        (by rw [Category.assoc, h.lift_snd, sq.w]), h.lift_snd _ _ _⟩
+
+/--
+```
+A ---u---> c₁.pt ---π---> X₁.obj j
+|           |                |
+g           |              f.app j
+|           |                |
+v           v                v
+B ---v---> c₂.pt ---π---> X₂.obj j
+```
+
+Given lifting problems indexed by `J` of the above form, construct a cone of `X₁` with point `B`
+whose components `B ⟶ X₁.obj j` are given by solutions of the lifting problems.
+
+-/
+@[simp]
+noncomputable
+def Limits.productOfShapeLiftingCone {J : Type*} {X₁ X₂ : Discrete J ⥤ C}
```

Comment:
I would suggest using the name `HasLiftingProperty.productLiftingCone` for this definition.

Reply:
(Actually, this definition and the next can be removed, see suggestions below.)